src/System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -380,6 +380,15 @@ public override string[] GetFileSystemEntries(string path, string searchPattern)
             return dirs.Union(files).ToArray();
         }
 
+        /// <inheritdoc />
+        public override string[] GetFileSystemEntries(string path, string searchPattern, SearchOption searchOption)
+        {
+            var dirs = GetDirectories(path, searchPattern, searchOption);
+            var files = GetFiles(path, searchPattern, searchOption);
+
+            return dirs.Union(files).ToArray();
+        }
+
```

Comment:
We already have a similar implementation in [`EnumerateFileSystemEntries`](https://github.com/TestableIO/System.IO.Abstractions/blob/bb3dde6e2be7acc7a76d22c717b4c554f0aa513e/src/System.IO.Abstractions.TestingHelpers/MockDirectory.cs#L654). What about calling `GetFileSystemEntries` from there for consistency?